scripts/startup/tar/linux/opensearch-tar-install.sh

```diff
@@ -10,7 +10,11 @@ cd $OPENSEARCH_HOME
 KNN_LIB_DIR=$OPENSEARCH_HOME/plugins/opensearch-knn/lib
 ##Security Plugin
 if [ -d "$OPENSEARCH_HOME/plugins/opensearch-security" ]; then
-        bash $OPENSEARCH_HOME/plugins/opensearch-security/tools/install_demo_configuration.sh -y -i -s
+        echo "OpenSearch 2.12.0 onwards, the security plugin introduces a change that requires an initial password for 'admin' user."
+        echo "Please define an environment variable 'initialAdminPassword' with a password string."
+        echo "Or create a file 'initialAdminPassword.txt' with a single line that contains the password string and place it under $OPENSEARCH_PATH_CONF folder."
+        echo "If none of these are provided, a password will be generated and printed out."
+        bash $OPENSEARCH_HOME/plugins/opensearch-security/tools/install_demo_configuration.sh -y -i -s || exit 1
```

Comment:
How do we handle the docker changes tho? As it uses the same demo scripts.
Especially that it is a bad idea to save a text file to have admin:admin as default.

Reply:
DEB/RPM as well.
This change here is simply exit out the startup if there is no password given, a breaking change.

Reply:
This PR focuses on TAR only. There will be subsequents PRs for different distributions.